ci/build-mingw64.sh

```diff
@@ -62,135 +75,184 @@ function makeplusinstall () {
+    fi
+}
+
+
+## mpv's dependencies
+
+_iconv () {
+    local ver=1.17
     gettar "https://ftp.gnu.org/pub/gnu/libiconv/libiconv-${ver}.tar.gz"
     builddir libiconv-${ver}
     ../configure --host=$TARGET $commonflags
     makeplusinstall
     popd
-fi
+}
+_iconv_mark=lib/libiconv.dll.a
 
-## zlib
-if [ ! -e "$prefix_dir/lib/libz.dll.a" ]; then
-    ver=1.2.13
+_zlib () {
+    local ver=1.3
     gettar "https://zlib.net/fossils/zlib-${ver}.tar.gz"
     pushd zlib-${ver}
     make -fwin32/Makefile.gcc clean
     make -fwin32/Makefile.gcc PREFIX=$TARGET- CC="$CC" SHARED_MODE=1 \
         DESTDIR="$prefix_dir" install \
         BINARY_PATH=/bin INCLUDE_PATH=/include LIBRARY_PATH=/lib
     popd
-fi
+}
+_zlib_mark=lib/libz.dll.a
 
-## ffmpeg
-if [ ! -e "$prefix_dir/lib/libavcodec.dll.a" ]; then
+_ffmpeg () {
     [ -d ffmpeg ] || $gitclone https://github.com/FFmpeg/FFmpeg.git ffmpeg
     builddir ffmpeg
     ../configure --pkg-config=pkg-config --target-os=mingw32 \
         --enable-cross-compile --cross-prefix=$TARGET- --arch=${TARGET%%-*} \
         --cc="$CC" --cxx="$CXX" $commonflags \
-        --disable-{doc,programs,muxers,encoders,devices}
+        --disable-{doc,programs,muxers,encoders} \
+        --enable-encoder=mjpeg,png
     makeplusinstall
     popd
-fi
+}
+_ffmpeg_mark=lib/libavcodec.dll.a
 
-## shaderc
-if [ ! -e "$prefix_dir/lib/libshaderc_shared.dll.a" ]; then
+_shaderc () {
     if [ ! -d shaderc ]; then
         $gitclone https://github.com/google/shaderc.git
         (cd shaderc && ./utils/git-sync-deps)
     fi
     builddir shaderc
-    cmake .. -DCMAKE_SYSTEM_NAME=Windows \
-        -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF \
-        -DSHADERC_SKIP_TESTS=ON -DCMAKE_INSTALL_PREFIX=/
+    cmake .. "${cmake_args[@]}" \
+        -DBUILD_SHARED_LIBS=OFF -DSHADERC_SKIP_TESTS=ON
     makeplusinstall
     popd
-fi
+}
+_shaderc_mark=lib/libshaderc_shared.dll.a
 
-## spirv-cross
-if [ ! -e "$prefix_dir/lib/libspirv-cross-c-shared.dll.a" ]; then
+_spirv_cross () {
     [ -d SPIRV-Cross ] || $gitclone https://github.com/KhronosGroup/SPIRV-Cross
     builddir SPIRV-Cross
-    cmake .. -DCMAKE_SYSTEM_NAME=Windows \
+    cmake .. "${cmake_args[@]}" \
         -DSPIRV_CROSS_SHARED=ON -DSPIRV_CROSS_{CLI,STATIC}=OFF
     makeplusinstall
     popd
-fi
+}
+_spirv_cross_mark=lib/libspirv-cross-c-shared.dll.a
+
+_vulkan_headers () {
+    [ -d Vulkan-Headers ] || $gitclone https://github.com/KhronosGroup/Vulkan-Headers
+    builddir Vulkan-Headers
+    cmake .. "${cmake_args[@]}"
+    makeplusinstall
+    popd
+}
+_vulkan_headers_mark=include/vulkan/vulkan.h
+
+_vulkan_loader () {
+    [ -d Vulkan-Loader ] || $gitclone https://github.com/KhronosGroup/Vulkan-Loader
+    builddir Vulkan-Loader
+    cmake .. "${cmake_args[@]}" \
+        -DENABLE_WERROR=OFF -DUSE_MASM=OFF
+    makeplusinstall
+    popd
+    sed -re '/libdir=/s|Lib(32)?|lib|' -i "$prefix_dir/lib/pkgconfig/vulkan.pc" # wat?
+}
+_vulkan_loader_mark=lib/libvulkan-1.dll.a
```

Comment:
This makes me think that it shouldn't be hard dependency for vulkan support and instead vulkan loader should be loaded dynamically on supported systems. Same as d3d11 and opengl.

Reply:
GPU drivers provide the Vulkan loader on Windows right? or only the ICDs?

> d3d11

no idea about this

> opengl

However for GL Microsoft provides the loader, namely [WGL](https://learn.microsoft.com/en-us/windows/win32/opengl/wgl-functions)

Reply:
> GPU drivers provide the Vulkan loader on Windows right? or only the ICDs?

Yes, they do. Of course not if you are using drivers from Vulkan era :)

Reply:
Will remove `vulkan-1.dll` from shipped DLLs.